config/src/main/java/org/axonframework/config/EventProcessingConfigurer.java

```diff
@@ -823,4 +823,30 @@ static DeadLetteringInvokerConfiguration noOp() {
             return (config, builder) -> builder;
         }
     }
+
+    /**
+     * Register the given {@code deadLetterProvider} as a default to build a {@link SequencedDeadLetterQueue} for
+     * {@link EventProcessor}s created in this configuration.
+     * <p>
+     * The {@code deadLetterProvider} is invoked only if a processing group is configured to have a dlq.
+     *
+     * @param deadLetterProvider a builder {@link Function} that provides a {@link SequencedDeadLetterQueue} for a
+     *                           processing group. It's possible to return null depending on the processing group, if so
+     *                           the next ones will be tried.
+     * @return the current {@link EventProcessingConfigurer} instance, for fluent interfacing
+     */
+    default EventProcessingConfigurer registerDeadLetterProvider(
+            BiFunction<Configuration, String, SequencedDeadLetterQueue<EventMessage<?>>> deadLetterProvider) {
```

Comment:
Nit: I think it's fair to rename `deadLetterProvider` to `deadLetterQueueProvider`